lib/src/screens/wallet_keys/wallet_keys_page.dart

```diff
@@ -262,6 +268,62 @@ class _WalletKeysPageBodyState extends State<WalletKeysPageBody>
       ),
     );
   }
+  
+
+  Widget? _buildPassphraseBox() {
+    if (widget.walletKeysViewModel.passphrase.isEmpty) return null;
+    return Container(
+      padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 14),
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(8),
+        color: Theme.of(context).cardColor,
+      ),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          Text(
+            S.of(context).passphrase_view_keys,
+            textAlign: TextAlign.center,
+            style: TextStyle(
+              fontSize: 14,
+              fontWeight: FontWeight.w500,
+              color: Theme.of(context).extension<CakeTextTheme>()!.buttonTextColor.withOpacity(0.5),
+            ),
+          ),
+          const SizedBox(width: 6),
+          Expanded(
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.spaceBetween,
+              children: [
+                Observer(builder: (BuildContext context) {
+                  return Text(
+                    (widget.walletKeysViewModel.obscurePassphrase) ?
+                      "*" * min(16, max(4, widget.walletKeysViewModel.passphrase.length)) :
```

Comment:
it's better not to show `*` based on the passphrase length, will fixate it at 5
```suggestion
                      "*****" :
```